shared/libraries/config_protocol/src/config_protocol_client.cpp

```diff
@@ -313,6 +324,24 @@ void ConfigProtocolClient::connect(const ComponentPtr& parent)
     if (!success)
         throw ConfigProtocolException("Protocol upgrade failed");
 
+    const auto localTypeManager = daqContext.getTypeManager();
+    const TypeManagerPtr typeManager = clientComm->sendCommand("GetTypeManager");
+    const auto types = typeManager.getTypes();
+
+    for (const auto& typeName : types)
+    {
+        const auto type = typeManager.getType(typeName);
+        if (localTypeManager.hasType(type.getName()))
+        {
+            const auto localType = localTypeManager.getType(type.getName());
+            if (localType != type)
+                throw InvalidValueException("Remote type different than local");
```

Comment:
I would probably just log this, and try to use the local types instead with this device. Then if they're incompatible, either throw an error later, or log that as well.

Throwing an exception while connecting seems like a bit overkill, as you can't handle the exception with an application-level wrapper.

Reply:
This will actually throw an exception inside "Connect" call. There are other reasons why Connect can throw exceptions. The client code / native config module should in any case wrap the connect call in try/catch statement and then rethrow or handle the error in addDevice call.